.github/workflows/release.yml

```diff
@@ -91,11 +92,26 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        python-version: [ '3.10']
-        os: [macos-latest, windows-latest, ubuntu-latest]
+        include:
+          - os: macos-latest
+            arch: x86_64
+            cibw_arch: "x86_64"
+          - os: macos-14  # https://github.com/actions/runner-images?tab=readme-ov-file#available-images
+            arch: arm64
+            cibw_arch: "arm64"
+          - os: windows-latest
+            cibw_arch: "auto"
+          - os: ubuntu-latest
+            cibw_arch: "auto aarch64"
+
     steps:
       - uses: actions/checkout@v3
 
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.10'
+
```

Comment:
What is the relationship between this python 3.10 setup step and the built wheels? Is 3.10 used to build wheels for all the versions? Are we only building wheels for 3.10?

Reply:
And why not 3.11?

Reply:
This python version will be used for setting up virtual environment and install dependencies including `cibuildwheel`. We used `3.10` back then because `cibuildwheel=2.4.0` only supported wheels for python versions up `3.10` so I used the same version as it's the latest version of python we supported to build wheels. 

To avoid the confusion, I updated the PR to use general `python3` command to let tasks use the default python versions that these images are using. For example, [MacOs 14](https://github.com/actions/runner-images/blob/main/images/macos/macos-14-arm64-Readme.md) is using python `3.11` and [Ubuntu-latest](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md) is using python `3.10`. But both versions are tested in our [main test workflow](https://github.com/amazon-ion/ion-python/pull/338/files#diff-7829468e86c1cc5d5133195b5cb48e1ff6c75e3e9203777f6b2e379d9e4882b3R33).

Reply:
A successful workflow without python 3.10 setup is available in my fork - https://github.com/cheqianh/ion-python/actions/runs/7891635687

Reply:
Well that still doesn't really answer the question. I dug into the docs and the answer is that the `CIBW` flags control  which architectures and python versions one builds for. I added a comment to the current skip setup because I think we need to be explicit about we _include_ not _exclude_ there. It would also make it clearer what is going on.

Reply:
Okay, will include this change in the next commit.

Reply:
>  Is 3.10 used to build wheels for all the versions? 

No, `cibuildwheel` will build wheels using different python versions respectively  

> Are we only building wheels for 3.10?

We are currently supporting the building of wheels for versions 3.8 to 3.11.